### ISSUE TITLE
self-defined tokenizers bug, maybe?

### ISSUE THREAD

**Issue by data-weirdo** [huggingface/tokenizers#766]
I don't know whether this is problem of 'WordLevelTrainer', 'Whitespace`, or 'tokenizers' library itself.

WordLevelTrainer does not reflect on the order I declarded.
For example, I declared as below: 
`trainer = WordLevelTrainer(special_tokens = ['[PAD]', '[CLS]', '[SEP]', '[UNK]', '[MASK]'],)`

And I identified that `tokenizer.json` file created from this code reflected on the right order as I intended as below:
` ...  {"type":"WordLevel","vocab":{"[PAD]":0,"[CLS]":1,"[SEP]":2,"[UNK]":3,"[MASK]":4, ... }`

However when I loaded this tokenizer by using a `PreTrainedTokenizerFast`,
I should have delcared this kind of statement first to use `pad_token_id` method,  

```
tokenizer.pad_token = '[PAD]'
tokenizer.cls_token = '[CLS]'
tokenizer.sep_token = '[SEP]'
tokenizer.unk_token = '[UNK]'
tokenizer.mask_token = '[MASK]'
tokenizer.pad_to_multiple_of = 8
```

and then, when I entered `tokenizer.pad_token_id`  , it should show 0, but it showed 3.  
Of course, I definitely used `tokenizer.enable_padding(pad_id=0, pad_token='[PAD]')` declaration at tokenizer training step.
Order of these five tokens were always fixed as below however I change the order of special tokens.  

```  
0: [UNK], 1: [CLS], 2: [SEP], 3: [PAD], 4: [MASK]
```

What's wrong? Do I understand sth wrong? As far as I tried, I think it is kind of a bug in transformer `tokenizers`!

**Comment by Narsil** [comment]
Hi @data-weirdo ,

I think everything is working as expected.

`tokenizers` has no semantic meaning for special tokens (cls, sep, or mask have no information carried with them).
`transformers` on the other hand has meaning for those (.pad_token, .cls_token, ...)

The way this is done, is with `config.json` or `special_tokens.json` which is used by transformers only (different from `tokenizer.json`). Check out an example here: https://huggingface.co/gpt2/blob/main/config.json#L9.

Basically `transformers is responsible for mapping semantic special tokens, to regular special tokens within `tokenizers`.

Is that clearer ?

**Comment by data-weirdo** [comment]
@Narsil Oh, I never know whether I could use `json` files on defining `tokenizers`.
Thank you so much for notifying it. I gonna look around the link you attache. Thank you :)
